File: app/src/main/java/com/gals/prayertimes/ui/MyMenuItemRecyclerViewAdapter.java
package com.gals.prayertimes.ui;

import androidx.recyclerview.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.gals.prayertimes.R;
import com.gals.prayertimes.ui.MenuItemFragment.OnListFragmentInteractionListener;
import com.gals.prayertimes.ui.SettingsMenuContent.SettingsMenuItem;

import java.util.List;

/**
 * {@link RecyclerView.Adapter} that can display a {@link SettingsMenuItem} and makes a call to the
 * specified {@link OnListFragmentInteractionListener}.
 */
public class MyMenuItemRecyclerViewAdapter extends RecyclerView.Adapter<MyMenuItemRecyclerViewAdapter.ViewHolder> {

    private final List<SettingsMenuItem> mValues;
    private final OnListFragmentInteractionListener mListener;

    public MyMenuItemRecyclerViewAdapter(List<SettingsMenuItem> items, OnListFragmentInteractionListener listener) {
        mValues = items;
        mListener = listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_menuitem, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.mItem = mValues.get(position);
        holder.mImage.setImageResource(mValues.get(position).img);
        holder.mNameView.setText(mValues.get(position).name);

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    Log.i("List", v.getId() + "  " + holder.mNameView);

                    mListener.onListFragmentInteraction(holder.mItem);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final ImageView mImage;
        public final TextView mNameView;
        public SettingsMenuItem mItem;

        public ViewHolder(View view) {
            super(view);
            mView = view;
            mImage = (ImageView) view.findViewById(R.id.item_image);
            mNameView = (TextView) view.findViewById(R.id.name);
        }

        @Override
        public String toString() {
            return super.toString() + " '" + mNameView.getText() + "'";
        }
    }


}

File: app/src/main/java/com/gals/prayertimes/ui/MyMenuItemRecyclerViewAdapter.kt
package com.gals.prayertimes.ui

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.gals.prayertimes.R
import com.gals.prayertimes.ui.MenuItemFragment.OnListFragmentInteractionListener
import com.gals.prayertimes.ui.SettingsMenuContent.SettingsMenuItem

/**
 * [RecyclerView.Adapter] that can display a [SettingsMenuItem] and makes a call to the
 * specified [OnListFragmentInteractionListener].
 */
class MyMenuItemRecyclerViewAdapter(
    private val mValues: List<SettingsMenuItem>,
    private val mListener: OnListFragmentInteractionListener?
) : RecyclerView.Adapter<MyMenuItemRecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(
                R.layout.fragment_menuitem,
                parent,
                false
            )
        return ViewHolder(view)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        holder.mItem = mValues[position]
        holder.mImage.setImageResource(mValues[position].img)
        holder.mNameView.text = mValues[position].name
        holder.mView.setOnClickListener { v ->
            if (mListener != null) {
                Log.i(
                    "List",
                    v.id.toString() + "  " + holder.mNameView
                )
                mListener.onListFragmentInteraction(holder.mItem)
            }
        }
    }

    override fun getItemCount(): Int {
        return mValues.size
    }

    inner class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView) {
        val mImage = mView.findViewById<View>(R.id.item_image) as ImageView
        val mNameView = mView.findViewById<View>(R.id.name) as TextView
        lateinit var mItem: SettingsMenuItem
        override fun toString(): String {
            return super.toString() + " '" + mNameView.text + "'"
        }
    }
}